modules/ui/sections/color_selection.js

```diff
@@ -0,0 +1,91 @@
+import { uiTooltip } from '../tooltip';
+import { uiCombobox } from '../combobox';
+import { uiIcon } from '../icon';
+import { uiSection } from '../section';
+import { utilNoAuto } from '../../util';
+
+
+export function uiSectionColorSelection(context) {
+  const l10n = context.systems.l10n;
+  const colors = context.systems.colors;
+  colors.on('colorsloaded', loadComboBoxData);
+  let comboData = [];
+
+  function loadComboBoxData(){
+    let colorSchemeKeys = Object.keys(colors.getAllColorSchemes());
+
+    for (let i = 0; i < colorSchemeKeys.length; i++) {
+      let colorObject = {};
+      let colorSchemeTitle = colorSchemeKeys[i];
+      colorObject.title = colorSchemeTitle;
+      colorObject.value = colorSchemeTitle;
+      comboData.push(colorObject);
+    }
+    return comboData;
+  }
+
+
+  const section = uiSection(context, 'preferences-color-selection')
+  .label(l10n.tHtml('preferences.color_selection.title'))
+  .disclosureContent(renderDisclosureContent);
+
+  const colorCombo = uiCombobox(context, 'color-selection');
+
+  let _colorSelectedId = null;
+  let _checkboxState = false;
+
+  function renderDisclosureContent(selection) {
+    // enter
+    let colorOptionsListEnter = selection.selectAll('.preferences-color-selection-list')
+      .data([0])
+      .enter()
+      .append('div')
+      .attr('class', 'layer-list preferences-color-selection-list')
+      .call(uiTooltip(context)
+        .title(l10n.tHtml('preferences.color_selection.tooltip')));
+
+
+    let pickerCombo = colorOptionsListEnter.append('div');
+    let options = ['default', 'red_blind', 'green_blind'];
+
+    pickerCombo.append('input')
+    .attr('class', 'color-select')
+      .attr('placeholder', l10n.t('preferences.color_selection.placeholder'))
+      .data(options)
+      .text(d => l10n.t(`preferences.color_selection.${d}`))
+      .call(utilNoAuto)
+      .call(colorCombo)
+      .on('blur change', d3_event => {
+        const element = d3_event.currentTarget;
+        const val = (element && element.value) || '';
+        const data = colorCombo.data();
+        if (data.some(item => item.value === val)) {
+          _colorSelectedId = val;
+
+          if (colors.currentColorScheme !== _colorSelectedId) {
+            colors.setColorScheme(_colorSelectedId);
+            context.scene().dirtyScene();
+            context.systems.map.deferredRedraw();
+          }
+
+        } else {
+          d3_event.currentTarget.value = '';
+          _colorSelectedId = null;
+        }
+      });
+
+    colorCombo.data(comboData);
```

Comment:
no changes necessary here, just set a breakpoint here and you should see the `comboData` has something like 

`{ title: 'default', value: 'Default'}, `{title:` 'red_blind', value: 'Protanopia'} etc.

Reply:
😄 
<img width="307" alt="Screenshot 2023-12-04 at 12 09 35 PM" src="https://github.com/facebook/Rapid/assets/95086147/8390d429-abae-4667-a56c-0e3fc6157d66">